homeassistant/components/light/mqtt.py

```diff
@@ -7,46 +7,85 @@
 import logging
 from functools import partial
 
+import voluptuous as vol
+
 import homeassistant.components.mqtt as mqtt
 from homeassistant.components.light import (
     ATTR_BRIGHTNESS, ATTR_RGB_COLOR, Light)
+from homeassistant.const import CONF_NAME
+import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.template import render_with_possible_json_value
-from homeassistant.util import convert
 
 _LOGGER = logging.getLogger(__name__)
 
+DEPENDENCIES = ['mqtt']
+
+CONF_STATE_TOPIC = 'state_topic'
+CONF_COMMAND_TOPIC = 'command_topic'
+CONF_STATE_VALUE_TEMPLATE = 'state_value_template'
+CONF_BRIGHTNESS_STATE_TOPIC = 'brightness_state_topic'
+CONF_BRIGHTNESS_COMMAND_TOPIC = 'brightness_command_topic'
+CONF_BRIGHTNESS_VALUE_TEMPLATE = 'brightness_value_template'
+CONF_RGB_STATE_TOPIC = 'rgb_state_topic'
+CONF_RGB_COMMAND_TOPIC = 'rgb_command_topic'
+CONF_RGB_VALUE_TEMPLATE = 'rgb_value_template'
+CONF_PAYLOAD_ON = 'payload_on'
+CONF_PAYLOAD_OFF = 'payload_off'
+CONF_OPTIMISTIC = 'optimistic'
+CONF_BRIGHTNESS_SCALE = 'brightness_scale'
```

Comment:
Optimistic can go to `const`, the others probably fit better in the MQTT component.